### DISCUSSION TITLE
Issue ideas - Milestone 2

### DISCUSSION THREAD

**Discussion by krigubir** [ksvedal/uu-chrome-extension#34] [Ideas]
Issue ideas relevant for milestone 2

**Comment by andreascnits** [comment]
aria-labels

Knapper som ikke har beskrivende tekst (meny-knapper, mest ikon/bilde-knapper sikkert) skal ha aria-label som sier hva knappen gjør / hva den heter

https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA14

**Reply by krigubir** [reply]
Tilleggsinfo: Punkt 4.1.2.b i WCAG sier at knapper skal være koblet til en ledetekst i koden. "Send inn skjema" er et eksempel på en ledetekst som beskriver knappens funksjon og som vil bli lest opp for brukeren.
Eksempel: button aria-label="Send inn skjema">Send</button

**Reply by krigubir** [reply]
Tilleggsinfo: Men, ledetekst behøver ikke være en aria-label! Det virker som ledetekst er alt som beskriver funksjonen til en knapp og som kan bli lest opp høyt for brukeren. Ledetekst kan derfor implementeres på flere måter: Knappen kan være intuitiv/selvbeskrivende ("click med to send this schema"), knappen kan ha en skjult ledetekst (som leses opp) ved bruk av CSS (visually hidden-tag, offscreen-tag), knappen kan beskrives ved en enkel paragraph-tag.

**Comment by krigubir** [comment]
Finnes det en global søkefunksjon som er tilgjengelig i alle sidene til nettsiden?
Poenget med en global søkefunksjon er at man enkelt skal kunne ta seg videre/tilbake/et annet sted uansett hvor på nettsiden man befinner seg.

ISSUE: er knappen en del av denne globale søkefunksjonen? Kan man navigere seg til en spesifikk knapp ved hjelp av søkefunksjonen?

**Comment by krigubir** [comment]
ISSUE: Har knappen en "title-attribute"? Denne funksjonen skal gi en beskrivende tekst til hva knappen gjør/hvor den tar deg. Ved å holde pilen over en knapp skal man kunne få opp en liten "popup-boks" som gir tilleggsinformasjon. F.eks: "klikk her for å sende inn skjemaet".

**Comment by kaechin** [comment]
Chrome-utvidelsen av OK-knappen når chatbot vises, reflekteres ikke.　
URL-en nedenfor er et eksempel.
https://coop.no/extra/
Hvis du ikke forstår hva jeg mener, spør meg direkte.

**Comment by kaechin** [comment]
Som vist i URL-en nedenfor, er knappen ikke omgitt av en ramme, og det er et designproblem.
https://www.posten.no/
Den gjeldende knappen er også rød, noe som gjør det vanskelig å se når nettstedet har rød bakgrunn. Derfor er det nødvendig å løse dette problemet.
Hvis du ikke forstår hva jeg mener, spør meg direkte.
